BASE/src/ServerTelemetryChannel/Implementation/TelemetryBuffer.cs

```diff
@@ -189,8 +194,10 @@ public virtual async Task FlushAsync()
                 // Flush on thread pull to offload the rest of the channel logic from the customer's thread.
                 // This also works around the problem in ASP.NET 4.0, does not support await and SynchronizationContext correctly.
                 // See also: http://www.bing.com/search?q=UseTaskFriendlySynchronizationContext
-                await Task.Run(() => this.serializer.Serialize(telemetryToFlush)).ConfigureAwait(false);
+                return await this.serializer.Serialize(telemetryToFlush, true).ConfigureAwait(false);
```

Comment:
Tip: async methods with some pre-amble and ending with a single "return await" still cause the compiler to emit the async state machine code. This can be simplified by removing the `async` from the method signature and returning tasks directly. It also means you don't need ConfigureAwait. i.e.
```csharp
public virtual Task<bool> ManualFlushAsync()
{
    List<ITelemetry> telemetryToFlush = this.MoveTelemeteryBufferForFlush();
    return telemetryToFlush != null ? this.serializer.Serialize(telemetryToFlush, true) : Task.FromResult(false);
}
```